src/vmm/src/vstate.rs

```diff
@@ -718,7 +719,7 @@ impl Vcpu {
                 // Emulation was interrupted, check external events.
                 Ok(VcpuEmulation::Interrupted) => break,
                 // Emulation errors lead to vCPU exit.
-                Err(_) => return StateMachine::next(Self::exited),
+                Err(_) => return self.exit(FC_EXIT_CODE_GENERIC_ERROR),
```

Comment:
In this case, you should change `run_emulation()` to not mark `KVM_EXIT_SHUTDOWN` as an error.
In case of `KVM_EXIT_SHUTDOWN`, we want `FC_EXIT_CODE_OK`.

Reply:
Yes. I did this in the last commit.